components/list/list.js

```diff
@@ -0,0 +1,71 @@
+import './list-item.js';
```

Comment:
Technically, this doesn't use `list-item` so maybe shouldn't be importing it? That means consumers will need to import it, but that might be good since in theory we might have other types of list items in the future that work with `d2l-list`.

Reply:
I wanted to make the import easier. Since they are forced to use this `list-item` as of right now.

I am not sure how I feel about having different types of `list-item`s since then it is easy to miss match. 

At which point should a new one be made? 
Should we have one for `selectors`? Should they all support the ability to select? 
Maybe the `list-items` take place of common content?

I started writing out all the different types and was very excited by the idea actually. Then I got to the one with basic content with the image inside element. Which I should be selectable which then fanned out too much and broke it.

Having multiple list-items might not be a bad idea I don't think it would be as messy as I originally thought and would be it rather module.

Reply:
Having different item components with a common mixin seemed to be helpful in dropdown menus, but @dbatiste lived and breathed that code so may feel differently.

Reply:
Yeah it was definitely helpful.  We were able to apply slotted styles by targeting `::slotted([role="menuitem"])`, `::slotted([role="menuitemcheckbox"])`, and `::slotted([role="menuitemradio"])`, which perhaps could be similarly done for list-items if we think we'll need different types of list items.  I think we need to figure out how the HM component fits into the DOM structure.

Reply:
So, I can rough out how the HM component would be used but honestly, I think I am too close to it. I'll ask Allan for his input on it.

Reply:
Here is what Allan mocked out:
```
class MyList {
    render() {
        return html`
            <d2l-list divider="top">
                ${this._activities.map(
                    activity => html`
                        <d2l-list-item>
                            <div class="d2l-list-item-text">
                                ${activity.name}
                            </div>
                            <div class="d2l-list-item-text-secondary">
                                ${`${activity.collection.children.length} Activities`}
                            </div>
                            <div slot="illustration">
                                <d2l-organization-image href=${activity.organization} />
                            </div>
                        </d2l-list-item>
                    `
                )}
            </d2l-list>
        `;
    }
}
```

Reply:
Although, he may be also too familiar with this.

Reply:
Wrapping the entire list in the HM component simplifies things.  So we would not expect a HM component per list-item (which complicates things a lot)?

Reply:
Yeah I really like this approach using `map` -- this is how I was hoping things would turn out.

Reply:
Oh good. So with the siren SDK doing it this way is so much easier now. That is also how I would code it up.

Reply:
For each list item, there could be actions, and they may not be the same for each item.  Is all that action info (for all items) going to come back in that one HM response?  I am a bit ignorant wrt Siren SDK - how does it help this?

Reply:
Siren-sdk lets us navigate the HM tree in a mostly clean way.

This allows us to get the information we need without making a HM component per list item. Then making more components based on different branches from there.